receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -121,14 +122,17 @@ type consumerGroupHandler struct {
 	unmarshaller Unmarshaller
 	nextConsumer consumer.TraceConsumer
 	ready        chan bool
+	readyCloser sync.Once
 
 	logger *zap.Logger
 }
 
 var _ sarama.ConsumerGroupHandler = (*consumerGroupHandler)(nil)
 
 func (c *consumerGroupHandler) Setup(session sarama.ConsumerGroupSession) error {
-	close(c.ready)
+	c.readyCloser.Do(func() {
```

Comment:
The setup can be called multiple times 

https://github.com/Shopify/sarama/issues/1793#issuecomment-683900202

> There is an issue with the example, as the Setup method could be called multiple times as part of when a member initially joins a group or a rebalance happens.

The example from sarama https://github.com/Shopify/sarama/blob/master/examples/consumergroup/main.go#L142 reinitializes the ready channel in the consumer loop that leads to data race. For us it's enough to close the ready channel once as it is used only once in receiver start method to block until receiver is ready to consumer mesages.